www/ruzo03/sp2/assets/js/recipes.js

```diff
@@ -0,0 +1,207 @@
+
+  const $filterCheckboxes = $filterBar.querySelectorAll("input:checked");
+
+  const queries = [];
+
+  if ($filterSearch.value) queries.push(["q", $filterSearch.value]);
+
+  if ($filterCheckboxes.length) {
+    for (const $checkbox of $filterCheckboxes) {
+      const key = $checkbox.parentElement.parentElement.dataset.filter;
+      queries.push([key, $checkbox.value]);
+    }
+  }
+
+  window.location = queries.length ? `?${queries.join("&").replace(/,/g, "=")}` : "/index.html";
+
+});
+
+$filterSearch.addEventListener("keydown", e => {
+  if (e.key === "Enter") $filterSubmit.click();
+});
+
+$filterClear.addEventListener("click", function () {
+
+  const $filterCheckboxes = $filterBar.querySelectorAll("input:checked");
+
+  $filterCheckboxes?.forEach(elem => elem.checked = false);
+  $filterSearch.value &&= "";
+
+});
+
+
+const queryStr = window.location.search.slice(1);
+const queries = queryStr && queryStr.split("&").map(i => i.split("="));
+
+const $filterCount = document.querySelector("[data-filter-count]");
+
+if (queries.length) {
+  $filterCount.style.display = "block";
+  $filterCount.innerHTML = queries.length;
+} else {
+  $filterCount.style.display = "none";
+}
+
+queryStr && queryStr.split("&").map(i => {
+  if (i.split("=")[0] === "q") {
+    $filterBar.querySelector("input[type='search']").value = i.split("=")[1].replace(/%20/g, " ");
+  } else {
+    $filterBar.querySelector(`[value="${i.split("=")[1].replace(/%20/g, " ")}"]`).checked = true;
+  }
+});
+
+const $filterBtn = document.querySelector("[data-filter-btn]");
+
+window.addEventListener("scroll", e => {
+  $filterBtn.classList[window.scrollY >= 120 ? "add" : "remove"]("active");
+});
+
+const  $gridList = document.querySelector("[data-grid-list]");
+const  $loadMore = document.querySelector("[data-load-more]");
+const  defaultQueries = [
+  ["mealType", "breakfast"],
+  ["mealType", "dinner"],
+  ["mealType", "lunch"],
+  ["mealType", "snack"],
+  ["mealType", "teatime"],
+  ...cardQueries
+];
+
+$gridList.innerHTML = $skeletonCard.repeat(20);
+let nextPageUrl = "";
+
+const renderRecipe = data => {
+
+  data.hits.map((item, index) => {
+
+    const {
+      recipe: {
+        image,
+        label: title,
+        totalTime: cookingTime,
+        uri
+      }
+    } = item;
+
+    const recipeId = uri.slice(uri.lastIndexOf("_") + 1);
+    const isSaved = window.localStorage.getItem(`cookio-recipe${recipeId}`);
+
+    const $card = document.createElement("div");
+    $card.classList.add("card");
+    $card.style.animationDelay = `${100 * index}ms`;
+
+    $card.innerHTML = `
+      <figure class="card-media img-holder">
+        <img src="${image}" width="195" height="195" loading="lazy" alt="${title}"
+          class="img-cover">
+      </figure>
+
+      <div class="card-body">
+
+        <h3 class="title-small">
+          <a href="./detail.html?recipe=${recipeId}" class="card-link">${title ?? "Untitled"}</a>

```

Comment:
aplpikace ma byt jedno strankova s dynamicky generovanym obsahem, nesmite to psat do vice stranek